### PR TITLE
enha: add run with importer e2e test

### PR DESCRIPTION
**Pull request by gabriel-aranha-cw**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Added a new case for `run-with-importer` in the `updateProviderUrl` function.
- Introduced a comprehensive integration test suite for `Run With Importer`, including tests for deploying and configuring the BRLC contract, long-duration transaction tests, and validation tests for transaction delays, balances, and transaction imports between Stratus and Run With Importer.
- Created a new GitHub Actions workflow to automate the execution of E2E tests with `Run With Importer`, including environment setup and test execution steps.
- Enhanced the justfile with new commands to facilitate running and managing E2E tests with `Run With Importer`, including commands for building binaries, starting services, and cleaning up after tests.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rpc.ts</strong><dd><code>Add `run-with-importer` case to `updateProviderUrl` function</code></dd></summary>
<hr>

e2e/cloudwalk-contracts/integration/test/helpers/rpc.ts

<li>Added a new case for <code>run-with-importer</code> in <code>updateProviderUrl</code> function.<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1522/files#diff-bfbd0a045ec4243ba68015b78cb7fb7cb485ff47e44bc3c22e3e155fc3ddf003">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>justfile</strong><dd><code>Add commands for E2E tests with `Run With Importer` in justfile</code></dd></summary>
<hr>

justfile

<li>Added new commands for running and managing E2E tests with <code>Run With </code><br><code>Importer</code>.<br> <li> Included commands for building binaries, starting services, and <br>cleaning up after tests.<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1522/files#diff-deb9bb56fb122db0b605aa5b63f95a4665c905b18dd670e1fa6c877576a94ff1">+72/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>importer.test.ts</strong><dd><code>Add integration tests for `Run With Importer`</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

e2e/cloudwalk-contracts/integration/test/importer.test.ts

<li>Added a new integration test suite for <code>Run With Importer</code>.<br> <li> Included tests for deploying and configuring BRLC contract.<br> <li> Added long-duration transaction tests with various parameters.<br> <li> Added validation tests for transaction delays, balances, and <br>transaction imports between Stratus and Run With Importer.<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1522/files#diff-4b834daff35390b2121977d1cde832c82bfe2e17937a70a7d2d02c44506d3487">+174/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>e2e-importer-online.yml</strong><dd><code>Add GitHub Actions workflow for E2E tests with `Run With Importer`</code></dd></summary>
<hr>

.github/workflows/e2e-importer-online.yml

<li>Created a new GitHub Actions workflow for E2E tests with <code>Run With </code><br><code>Importer</code>.<br> <li> Defined triggers for pull requests and pushes to the main branch.<br> <li> Included steps for setting up the environment and running the tests.<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1522/files#diff-b05519f982fd85c4b17b6beff9813388c13cabc6fb19da1160bfa1caf0d50fe4">+86/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -298,6 +298,76 @@ e2e-flamegraph:
...
+    #!/bin/bash
+
+    # Build Stratus and Run With Importer binaries
+    echo "Building Stratus and Run With Importer binaries"
+    cargo build --release --bin stratus --bin run-with-importer --features dev
+
+    mkdir e2e_logs
+
+    # Wait for builds to finish
+    wait
```
Comment (mayconamaroCW) [inline justfile:322]: cargo build is a blocking job, I don't think this wait is needed

File: .github/workflows/e2e-importer-online.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+name: E2E Run With Importer
```
[the remark this answers is not in the document]
Comment (gabriel-aranha-cw) [reply .github/workflows/e2e-importer-online.yml]: Done.

File: .github/workflows/e2e-importer-online.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      - '.github/workflows/pr-agent.yml'
+      - '.github/workflows/build-binary.yml'
+      - '.github/CODEOWNERS'
+      - 'config/**'
+      - 'README.md'
+      - 'LICENSE'
+      - 'CONTRIBUTING.md'
+      - 'utils/slack-notifiers/**'
+  workflow_dispatch:
+  push: # Rebuild cache policy
```
[the remark this answers is not in the document]
Comment (gabriel-aranha-cw) [reply .github/workflows/e2e-importer-online.yml]: Done.

File: justfile (hunk 1 above)
Comment (gabriel-aranha-cw) [reply justfile]: Done.

**Pull request merged**